apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -35,32 +35,37 @@ public class TraceTest
     {
         private readonly TraceEntryPolling _polling = new TraceEntryPolling();
 
-        private readonly TestServer _noBufferHighQps = new TestServer(
-            new WebHostBuilder().UseStartup<TraceTestNoBufferHighQpsApplication>());
-
-        private readonly TestServer _noBufferLowQps = new TestServer(
-            new WebHostBuilder().UseStartup<TraceTestNoBufferLowQpsApplication>());
+        public TraceTest()
+        {
+            // The rate limiter instance is static and only set once.  If we do not reset it at the
+            // beginning of each tests the qps will not change.  This is dependent on the tests not
+            // running in parallel.
+            RateLimiter._instance = null;
```

Comment:
Yea its fairly confusing using the constructor for a before each, but that how xunit works.

https://xunit.github.io/docs/shared-context.html